### PR TITLE
Update sqlite coverage

### PR DESCRIPTION
**Pull request by kbenne**
Mark's edit:

This is a major refactor of SQLite. This was done to facilitate unit testing.

This refactoring also caught a few existing bugs in the SQLite. 

One major bug, which required a schema change, was in ZoneList and ZoneGroup tables. The ZoneList table only output 1 Zone associated with ZoneList when there could be N number of Zones. Now the ZoneList table has a many-to-many relationship with Zones table by a new join table, ZoneInfoZoneLists.

ZoneGroup now includes the foreign key index of the associated ZoneList, which was missing previously.

[OtherEquipment error](https://github.com/NREL/EnergyPlus/pull/4680/files#diff-f3feb23447c269b298234c8ed8690c2fL781)

@kbenne @Myoldmopar @macumber 

[#87522778]

### CODE REVIEW

**Comment by mbadams5** [comment]
@kbenne @Myoldmopar @macumber 

This is ready for review now. The regression diffs are explainable and expected.

**Comment by kbenne** [comment]
@mbadams5 can you summarize the changes, particularly schema changes.  Here is what I see so far.

- Created Sqlite Factory that gets input and the constructs class, passing input as arguments to ctor.
- Removed use of global data in favor of method arguments
- Minor schema changes, including zone lists and groups, maybe others.

File: tst/EnergyPlus/unit/SQLite.unit.cc [hunk 1]
```diff
@@ -0,0 +1,1055 @@
...
+		std::string timestepTypeName( const int timestepType ) {
+			return sqlite_test->timestepTypeName( timestepType );
+		}
+
+		std::string reportingFreqName( const int reportingFreqIndex ) {
+			return sqlite_test->reportingFreqName( reportingFreqIndex );
+		}
+
+		std::string columnText( const unsigned char* column ) {
+			return std::string( reinterpret_cast<const char*>( column ) );
```
Comment (kbenne) [inline tst/EnergyPlus/unit/SQLite.unit.cc]: weird ???

Comment (mbadams5) [inline tst/EnergyPlus/unit/SQLite.unit.cc]: I actually copied this from [OpenStudio](https://github.com/NREL/OpenStudio/blob/23ab299e80c80c9d57a857e6300b3701340d2f76/openstudiocore/src/utilities/sql/SqlFile_Impl.cpp#L48) ...

File: tst/EnergyPlus/unit/SQLite.unit.cc [hunk 2]
```diff
@@ -0,0 +1,1055 @@
...
+
+		std::string storageType( const int storageTypeIndex ) {
+			return sqlite_test->storageType( storageTypeIndex );
+		}
+
+		std::string timestepTypeName( const int timestepType ) {
+			return sqlite_test->timestepTypeName( timestepType );
+		}
+
+		std::string reportingFreqName( const int reportingFreqIndex ) {
```
Comment (kbenne) [inline tst/EnergyPlus/unit/SQLite.unit.cc]: Why do these methods exist in the test fixture to begin with?  Access policy maybe.  Did you friend the test fixture or something?  I recall some discussion we had about this issue.  don't remember where it landed.

File: src/EnergyPlus/SQLiteProcedures.cc [hunk 3]
```diff
@@ -2060,442 +1992,768 @@ void SQLite::updateSQLiteSimulationRecord(
...
-		sqliteBindForeignKey(m_nominalSteamEquipmentInsertStmt, 4, steamEquipHB.SchedPtr);
-		sqliteBindDouble(m_nominalSteamEquipmentInsertStmt, 5, steamEquipHB.DesignLevel);
-		sqliteBindDouble(m_nominalSteamEquipmentInsertStmt, 6, steamEquipHB.FractionLatent);
-		sqliteBindDouble(m_nominalSteamEquipmentInsertStmt, 7, steamEquipHB.FractionRadiant);
-		sqliteBindDouble(m_nominalSteamEquipmentInsertStmt, 8, steamEquipHB.FractionLost);
-		sqliteBindDouble(m_nominalSteamEquipmentInsertStmt, 9, steamEquipHB.FractionConvected);
-		sqliteBindText(m_nominalSteamEquipmentInsertStmt, 10, steamEquipHB.EndUseSubcategory);
+void SQLite::addZoneData( int const number, DataHeatBalance::ZoneData const & zoneData ) {
+	zones.push_back(
+		std::unique_ptr<Zone>(new Zone(m_errorStream, m_db, number, zoneData))
```
Comment (kbenne) [inline src/EnergyPlus/SQLiteProcedures.cc]: The way you are doing this, one Zone instance per db record looks like Active Record pattern to me.  Is that what you are going for?  Why is the insertIntoSqlite a different step?  That is it seems you defer the actually committing to db.

**Comment by kbenne** [comment]
@mbadams5 Can you update the Output Details and Examples Document with a snippet of the changes here https://github.com/NREL/EnergyPlusBuildSupport/tree/master/docs/src/mods/for_2015_03_release ? There are descriptions of the database tables that need to be updated to reflect your (minor) schema changes.
